xbmc/utils/URIUtils.cpp

```diff
@@ -229,7 +232,7 @@ void URIUtils::Split(const std::string& strFileNameAndPath,
     if (ch == '/' || ch == '\\' || (ch == ':' && i == 1)) break;
     else i--;
   }
-  if (i == 0)
+  if (i == 0 && strFileNameAndPath[0] != '/' && strFileNameAndPath[0] != '\\')
```

Comment:
Wouldn't it be clearer to change the the condition of the while above to `i >= 0` and then to delete this entire condition?